### DISCUSSION TITLE
How to dynamic data properties in form

### DISCUSSION THREAD

**Discussion by madsem** [alpinejs/alpine#3161] [1. Help]
I'm building a form wizard, where individual steps are supposed to re-use form elements.

So I'm trying to be able to use form elements, and dynamically push the values into an object that exists in my Alpine component:

```html
<div class="box-flex wrap flex-center not-prose">
    {%- for answer in steps.answers -%}
    <label for="{{ answer.value }}" class="cursor-pointer w-full flex justify-center items-center bg-sky-900 text-white no-underline font-semibold shadow-md text-lg rounded-md my-2 h-16">
        {{- answer.label | markdownify | tokenReplacer | safe -}}
        <input type="radio"
            name="{{ steps.input_name }}"
            id="{{- answer.value -}}"
            x-show="!completed"
            x-on:click.prevent="forward()"
            class="hidden"
            value="{{- answer.value -}}"
            x-model="formData.metaData.{{ steps.input_name }}"
            {% if steps.checked_value == answer.value %}checked{% endif%}>
        </label>
    {%- endfor -%}
</div>
```

My component data:
```js
formData: {
    fname: "",
    lname: "",
    email: "",
    street: "",
    phone: "",
    zip: "",
    city: "", 
    state: "", 
    country: "",
    metaData: {},
  },
```

Expected (Or ideal) result would be that the component above, watches and updates the `formData.metaData.xxx` property as it does with all the others. But whenever I post the form `metaData` is just an empty string.

I'm sure this is an easy thing to do with Alpine, and that I'm a little dum-dum :)

Hope that someone can help me figuring this out!

**Comment by ekwoka** [comment]
I believe this is an issue with how form data handles that information.

I'm not positive it can handle nested objects so easily.

Could try posting it as json isntead of formdata.

**Reply by madsem** [reply]
Problem is, that Alpine doesn't even fill these dynamic, nested properties inside `metaData`, so they just don't exist even before the actual form submit.
